### ISSUE TITLE
Lazy loading

### ISSUE THREAD

**Issue by thomaxxl** [femioladeji/react-slideshow#59]
I like this package but it becomes slow with a large amount of images.
Is it possible to lazy load the images in the slideshow?

**Comment by femioladeji** [comment]
@thomaxxl thanks for pointing that out. I'll look into that

**Comment by femioladeji** [comment]
Hi @thomaxxl what you can do for now is to do something like this. The function lazyLoadImages will lazy load all `img` tag with class `lazy`. It's based on this [article](https://developers.google.com/web/fundamentals/performance/lazy-loading-guidance/images-and-video).

I'll look into a more robust way to handle this for all browsers but for now, this will work for you
```
import React, { Component } from "react";
import { Slide } from "react-slideshow-image";
import "./styles.css";

const lazyLoadImages = () => {
  const lazyImages = [].slice.call(document.querySelectorAll("img.lazy"));
    if ("IntersectionObserver" in window) {
      let lazyImageObserver = new IntersectionObserver(function(
        entries,
        observer
      ) {
        entries.forEach(function(entry) {
          if (entry.isIntersecting) {
            let lazyImage = entry.target;
            lazyImage.src = lazyImage.dataset.src;
            lazyImage.classList.remove("lazy");
            lazyImageObserver.unobserve(lazyImage);
          }
        });
      });

      lazyImages.forEach(function(lazyImage) {
        lazyImageObserver.observe(lazyImage);
      });
    }
}
class App extends Component {
  componentDidMount() {
    lazyLoadImages()
  }

  render() {
    const properties = {
      duration: 5000,
      autoplay: false,
      transitionDuration: 500,
      infinite: true,
      indicators: true
    };
    const slideImages = [
      "https://images.unsplash.com/photo-1509721434272-b79147e0e708?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1500&q=80",
      "https://images.unsplash.com/photo-1506710507565-203b9f24669b?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1536&q=80",
      "https://images.unsplash.com/photo-1536987333706-fc9adfb10d91?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1500&q=80",
      "https://images.unsplash.com/photo-1444525873963-75d329ef9e1b?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1500&q=80"
    ];
    return (
      <div className="App">
        <h3>Slide Effect</h3>
        <div className="slide-container">
          <Slide {...properties}>
            {slideImages.map((each, index) => (
              <div key={index} className="each-slide">
                <img
                  className="lazy"
                  style={{height: '300px', width: '100%', 'object-fit': 'cover'}}
                  src="data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs="
                  data-src={each}
                  alt="sample"
                />
              </div>
            ))}
          </Slide>
        </div>
      </div>
    );
  }
}

export default App;

```

**Comment by thomaxxl** [comment]
Thanks, didn't have the time tot test it though.

**Comment by thomaxxl** [comment]
Works great, thanks!

**Comment by thomaxxl** [comment]
Hi, the indicator bar disappears when a lot of images are in the slideImages list.
Should I create a custom bar with goTo() to handle this or is there another way?

**Comment by femioladeji** [comment]
No,  it shouldn't disappear Is it more than this [https://codesandbox.io/s/snowy-water-ldeym](https://codesandbox.io/s/snowy-water-ldeym)?

It should show all the indicators. Can I see a screenshot of what it looks like

**Comment by thomaxxl** [comment]
Hmm, in my component the dots are stretched until they fade away.
However in your example the dots appear on new rows.

I'm gonna check mycode.
Thanks a lot anyway!

**Comment by femioladeji** [comment]
Check if you're using the latest version of the package also. I fixed the issue in this PR #52

**Comment by thomaxxl** [comment]
Ah, this works better, thanks again :)
It would be nice to have the indicators to be customizable as well, adding hovers, colorize individual indicators, intervals etc.

Anyway, I'm happy how it works now, great work.
